Xamarin.Forms.Platform.iOS/CollectionView/ObservableGroupedSource.cs

```diff
@@ -130,53 +127,49 @@ void ResetGroupTracking()
 			}
 		}
 
-		async void CollectionChanged(object sender, NotifyCollectionChangedEventArgs args)
+		void CollectionChanged(object sender, NotifyCollectionChangedEventArgs args)
 		{
 			if (Device.IsInvokeRequired)
 			{
-				await Device.InvokeOnMainThreadAsync(async () => await CollectionChanged(args));
+				Device.BeginInvokeOnMainThread(() => CollectionChanged(args));
 			}
 			else
 			{
-				await CollectionChanged(args);
+				CollectionChanged(args);
 			}
 		}
 
-		async Task CollectionChanged(NotifyCollectionChangedEventArgs args)
+		void CollectionChanged(NotifyCollectionChangedEventArgs args)
```

Comment:
I think there are some issues related to `"SIGABRT: Index was out of range. Must be non-negative and less than the size of the collection. Parameter name: index"`, I wonder if this change can impact on those issues. We could do a quick review of them after merging this PR.